module_test.go

```diff
@@ -241,6 +243,187 @@ func TestModuleSuccess(t *testing.T) {
+				assert.Equal(t, tt.wantEventTypes, spy.EventTypes())
+
+				spy.Reset()
+				app.RequireStart().RequireStop()
+
+				require.NoError(t, app.Err())
+
+				assert.Equal(t, []string{"Started", "Stopped"}, spy.EventTypes())
+			})
+		}
+	})
+
+	type NamedSpy struct {
+		fxlog.Spy
+		name string
+	}
+
+	t.Run("decorator on module logger does not affect app logger", func(t *testing.T) {
+		t.Parallel()
+
+		appSpy := NamedSpy{name: "app"}
+		moduleSpy := NamedSpy{name: "redis"}
+
+		redis := fx.Module("redis",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "redis"}
+			}),
+			fx.Supply(&appSpy),
+			fx.Replace(&moduleSpy),
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "redis", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		app := fxtest.New(t,
+			redis,
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "app", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		assert.Equal(t, []string{
+			"Provided", "Supplied", "Replaced",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, moduleSpy.EventTypes())
+
+		assert.Equal(t, []string{
+			"Provided", "Provided", "Provided",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, appSpy.EventTypes())
+
+		appSpy.Reset()
+		app.RequireStart().RequireStop()
+
+		require.NoError(t, app.Err())
+
+		assert.Equal(t, []string{"Started", "Stopped"}, appSpy.EventTypes())
+	})
+
+	t.Run("module uses parent module's logger to log events", func(t *testing.T) {
+		t.Parallel()
+
+		appSpy := NamedSpy{name: "app"}
+		childSpy := NamedSpy{name: "child"}
+
+		grandchild := fx.Module("grandchild",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "grandchild"}
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "grandchild", r.Name)
+			}),
+		)
+
+		child := fx.Module("child",
+			grandchild,
+			fx.Supply(&appSpy),
+			fx.Replace(&childSpy),
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "child", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "grandchild", r.Name)
+			}),
+		)
+
+		app := fxtest.New(t,
+			child,
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "app", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "grandchild", r.Name)
+			}),
+		)
+
+		// events from grandchild also logged in child logger
+		assert.Equal(t, []string{
+			"Supplied", "Provided", "Replaced", "LoggerInitialized",
+			//Invoke logged twice, once from child and another from grandchild
+			"Invoking", "Invoked", "Invoking", "Invoked",
+		}, childSpy.EventTypes())
+
+		// events from modules do not appear in app logger
+		assert.Equal(t, []string{
+			"Provided", "Provided", "Provided",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, appSpy.EventTypes())
```

Comment:
Add what you have as a comment as the last parameter of assert.Equal and it'll show up in the test failure message if it passes. i.e.
```suggestion
		// events from modules do not appear in app logger
		assert.Equal(t, []string{
			"Provided", "Provided", "Provided",
			"LoggerInitialized", "Invoking", "Invoked",
		}, appSpy.EventTypes(), "events from modules should not appear in app logger")
```